lumicks/pylake/correlated_stack.py

```diff
@@ -340,6 +340,59 @@ def select_frame(event):
 
         fig.canvas.mpl_connect('button_press_event', select_frame)
 
+    def export_tiff(self, file_name, roi=None):
+        """Export a video of a particular scan plot
+
+        Parameters
+        ----------
+        file_name : str
+            File name to export to.
+        roi : list_like
+            region of interest in pixel values [xmin, xmax, ymin, ymax]
+        """
+        from . import __version__ as version
+
+        def parse_tags(frame):
+            # Parse original file tags into list of tuples [(code, dtype, count, value, writeonce)]
+            # Only tags that are not resolved automatically by `TiffWriter.save()` are needed
+            n_channels = 3 if frame.is_rgb else 1
+            datetime = frame._src.tags["DateTime"].value
+            output_tags = ((274, "H", 1, 1),                            # Orientation
+                           (339, "H", n_channels, (1, )*n_channels),    # SampleFormat
+                           (306, "s", len(datetime), datetime))         # DateTime
+            return output_tags
+
+        # get ROI coordinates
+        if roi is not None:
+            xmin, xmax, ymin, ymax = roi
+        else:
+            ymin, xmin = 0, 0
+            ymax = self._get_frame(0).data.shape[0]
+            xmax = self._get_frame(0).data.shape[1]
+
+        # re-name alignment matrices fields in image description
+        # to reflect the fact that the image has already been processed
+        description = self._get_frame(0)._src.description
+        if description != "":
```

Comment:
Is the `!= ""` needed here?

Reply:
I had some test data where the description tag was missing, and therefore returned as an empty string when opened with `tifffile`. Without the check, you get an exception with `json.loads()`. I don't know that this would ever actually happen from a stable version of Bluelake though.

Reply:
No, don't get rid of the check. I meant that an empty string usually behaves as `False` when used as a conditional statement.